pkg/debuginfo/cache.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debuginfo
+
+import (
+	"github.com/parca-dev/parca-agent/pkg/objectfile"
+)
+
+type cache struct {
+	files map[string]*debugInfoFile
+}
+
+func newCache() *cache {
+	return &cache{
+		files: make(map[string]*debugInfoFile),
+	}
+}
+
+func (c *cache) debugInfoFile(objFile *objectfile.ObjectFile) *debugInfoFile {
```

Comment:
it's quite confusing to me as the reader that this function is both used to read and write from the cache, I think making this a read-through cache would be easier to understand for the user, maybe even wrapping the gRPC client, so the caching logic doesn't spill into the business logic

Reply:
This is actually a read-through cache. The idea is to reduce the redundant work of find the debug information and maybe extraction. So it's not directly related to gRPC calls. This is one layer above. I realized I can do a better job of explaining this and possibly improve the interface. I'll do so.